packages/@webex/plugin-cc/src/AgentConfigService/types.ts

```diff
@@ -0,0 +1,232 @@
+/**
+ * Represents the request to UserRequest.
+ *
+ * @public
+ */
+
+export interface UserRequest {
+  /**
+   * Org id of the User
+   */
+  orgId: string;
+
+  /**
+   * CI user id of the User
+   */
+  ciUserId: string;
+}
+
+/**
+ * Represents the response from UserResponse.
+ *
+ * @public
+ */
+
+export interface UserResponse {
+  /**
+   * Identifier for a Desktop Profile.
+   */
+  agentProfileId: string;
+
+  /**
+   * This Specify the teams id which got assigned to the user.
+   */
+  teamIds: string[];
+
+  /**
+   * Identifier for an user profile which a Contact Center administrator has configured.
+   */
+  userProfileId: string;
+}
+
+/**
+ * Represents the request to DesktopProfileRequest.
+ *
+ * @public
+ */
+
+export interface DesktopProfileRequest {
+  /**
+   * ID of the Organization containing the Desktop Profile to be retrieved.
+   */
+  orgId: string;
+
+  /**
+   * ID of the Desktop Profile to be retrieved.
+   */
+  desktopProfileId: string;
+}
+
+/**
+ * Represents the response from DesktopProfileResponse.
+ *
+ * @public
+ */
+export interface DesktopProfileResponse {
+  /**
+   * Represents the voice options of an agent.
+   */
+
+  loginVoiceOptions: string[];
+
+  /**
+   * Specify the wrap-up codes that the agents can select when they wrap up a contact.
+   */
+
+  accessWrapUpCode: string;
```

Comment:
What is the difference between accessWrapUpConde and wrapUpCode? Same question for idleCodes and why do we have different conditions for each of these implemented in the code

Reply:
I have updated the description. We need to filter out wrapUpCodes and idleCodes based on accessWrapUpCode & accessIdleCode, here ALL specify the wrap-up codes that the agents can select when they wrap up a contact.